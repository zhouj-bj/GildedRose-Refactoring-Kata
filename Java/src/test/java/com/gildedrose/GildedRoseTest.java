package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] { new Item("Conjured Mana Cake", 0, 2) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals(0, app.items[0].quality);
    }

    @Test
    void testAgedBrie() {
        Item[] items = new Item[] { new Item("Aged Brie", 0, 10) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals(12, app.items[0].quality);
    }

    @Test
    public void testApproval() {
        CombinationApprovals.verifyAllCombinations(
            this::callUpdateQuality,
            new String[]{"foo", "Sulfuras, Hand of Ragnaros", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"},
            new Integer[]{0, -1, 13},
            new Integer[]{0, 1, 50, 49}
        );
    }

    private String callUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateQuality();

        return gildedRose.items[0].toString();
    }

}
