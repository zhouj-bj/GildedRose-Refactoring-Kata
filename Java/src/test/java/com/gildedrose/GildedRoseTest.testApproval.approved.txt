[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 50] => foo, -1, 48 
[foo, 0, 49] => foo, -1, 47 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 50] => foo, -2, 48 
[foo, -1, 49] => foo, -2, 47 
[foo, 13, 0] => foo, 12, 0 
[foo, 13, 1] => foo, 12, 0 
[foo, 13, 50] => foo, 12, 49 
[foo, 13, 49] => foo, 12, 48 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, 13, 0] => Sulfuras, Hand of Ragnaros, 13, 0 
[Sulfuras, Hand of Ragnaros, 13, 1] => Sulfuras, Hand of Ragnaros, 13, 1 
[Sulfuras, Hand of Ragnaros, 13, 50] => Sulfuras, Hand of Ragnaros, 13, 50 
[Sulfuras, Hand of Ragnaros, 13, 49] => Sulfuras, Hand of Ragnaros, 13, 49 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, 13, 0] => Aged Brie, 12, 1 
[Aged Brie, 13, 1] => Aged Brie, 12, 2 
[Aged Brie, 13, 50] => Aged Brie, 12, 50 
[Aged Brie, 13, 49] => Aged Brie, 12, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 13, 0] => Backstage passes to a TAFKAL80ETC concert, 12, 1 
[Backstage passes to a TAFKAL80ETC concert, 13, 1] => Backstage passes to a TAFKAL80ETC concert, 12, 2 
[Backstage passes to a TAFKAL80ETC concert, 13, 50] => Backstage passes to a TAFKAL80ETC concert, 12, 50 
[Backstage passes to a TAFKAL80ETC concert, 13, 49] => Backstage passes to a TAFKAL80ETC concert, 12, 50 
